### DISCUSSION TITLE
Running `now dev` gives error 'not found: next'

### DISCUSSION THREAD

**Discussion by pranjaljately** [vercel/next.js#13598] [Help]
Created a nextjs-blog as per the docs. Can download the code from [here](https://nextjs.org/learn/basics/deploying-nextjs-app/setup)

I am trying to run `now dev` and keep getting the following error: **not found: next**

![image](https://user-images.githubusercontent.com/30666375/83339008-91662600-a2c1-11ea-9a61-92ab0583a113.png)

**OS**: Windows 10 
**Now CLI version**: 19.0.1

**List of project dependencies:** 
- cookie@0.4.1
- date-fns@2.14.0
- gray-matter@4.0.2
- next@9.4.0
- react@16.13.1
- react-dom@16.13.1
- remark@12.0.0
- remark-html@11.0.2

**Comment by todortotev** [comment]
Can you please provide a repository to clone so I can reproduce the error?

**Comment by pranjaljately** [comment]
Repo: https://github.com/pranjaljately/nextjs-blog-temp
Or run: `npm init next-app nextjs-blog-temp --example "https://github.com/vercel/next-learn-starter/tree/master/basics-final`

Thanks!

**Comment by pranjaljately** [comment]
I should probably add that I have downloaded and installed the Vercel CLI but same issue when running `vercel dev`

![image](https://user-images.githubusercontent.com/30666375/83340853-ee1f0c00-a2d4-11ea-96f5-a6788db58e78.png)

**Comment by paulogdm** [comment]
I could not reproduce the issue:

```
▲ nextjs-blog-temp [master] npm i
npm WARN deprecated mkdirp@0.5.3: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN notsup Unsupported engine for watchpack-chokidar2@2.0.0: wanted: {"node":"<8.10.0"} (current: {"node":"12.17.0","npm":"6.14.5"})
npm WARN notsup Not compatible with your version of node/npm: watchpack-chokidar2@2.0.0
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 865 packages from 347 contributors and audited 867 packages in 24.245s

82 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
▲ nextjs-blog-temp [master] vercel    
Vercel CLI 19.0.1
? Set up and deploy “~/Codes/tmp/nextjs-blog-temp”? [Y/n] y
? Which scope do you want to deploy to? Paulo Guarnier De Mitri
? Link to existing project? [y/N] n
? What’s your project’s name? nextjs-blog-temp
? In which directory is your code located? ./
Auto-detected project settings (Next.js):
- Build Command: `npm run build` or `next build`
- Output Directory: Next.js default
- Development Command: next dev --port $PORT
? Want to override the settings? [y/N] n
🔗  Linked to paulo/nextjs-blog-temp (created .vercel)
🔍  Inspect: https://vercel.com/paulo/nextjs-blog-temp/dc2ygfrec [3s]
✅  Production: https://nextjs-blog-temp-ten.now.sh [copied to clipboard] [41s]
📝  Deployed to production. Run `now --prod` to overwrite later (https://zeit.ink/2F).
💡  To change the domain or build command, go to https://zeit.co/paulo/nextjs-blog-temp/settings
▲ nextjs-blog-temp [master] vercel dev
Vercel CLI 19.0.1 dev (beta) — https://vercel.com/feedback
> Running Dev Command “next dev --port $PORT”
[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3000 ...
Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade`
> Ready! Available at http://localhost:3000
```

**Reply by pranjaljately** [reply]
Did you happen to try this on a Windows machine? Seems to only be an issue on Windows...

**Comment by dan-turner** [comment]
Having the same problem here...  Exact same commit that worked this morning on the MacBooks of our dev team, no longer works this evening.

You guys really are living "the move fast and break things" motto...

**Comment by dan-turner** [comment]
We use a monorepo approach.

Our approx vercel.json:
```
{
  "version": 2,
  "builds": [
    {
      "src": "api/src/routes/**/*.ts",
      "use": "@now/node",
      "config": {
        "includeFiles": ["api/config/*"]
      }
    },
    { "src": "web/next.config.js", "use": "@now/next" }
  ],
  "routes": [
    { "src": "/api/(.*)", "dest": "/api/src/routes/$1" },
    { "src": "/(.*)", "dest": "/web/$1" }
  ]
}
```

The root has it's own package.json:
```
"devDependencies": {
    "vercel": "19.0.0"
  },
```

`/web/` has it's own package.json:
```
"dependencies": {
    <snipped>
    "next": "9.4.2",
    <snipped>
    "react": "16.13.1",
    <snipped>
  },
```

`/api/` has it's own package.json (irrelevant contents):

Output from `vercel dev`:

```
> Running Dev Command “next dev --port $PORT”
> [debug] [2020-06-17T09:29:32.992Z] Starting dev command with parameters : {"cwd":"/Users/dan/code/tex/app","command":"next dev --port 59247","port":59247}
> [debug] [2020-06-17T09:29:32.994Z] Spawning dev command: npx --no-install next dev --port 59247
not found: next
```

All of a sudden vercel is now trying to run `next` from the root of the monorepo instead of inside the `/web` directory where next is installed/used.

**Reply by dan-turner** [reply]
If I delete the `web` build and `web` route I **_still_** see `not found: next` in the output.  Even if I physically delete the whole `web` directory?

**Reply by dan-turner** [reply]
After `ctrl-c` out of the busted command, minutes later I see this stuff appear in the terminal:

```
› Error! Detecting port 51451 timed out after 300000ms
Error! Detecting port 51670 timed out after 300000ms
```

`ps aux | grep vercel` also shows some stray vercel processes that don't get terminated

**Comment by dan-turner** [comment]
@pranjaljately we managed to "fix" the issue.  The problem, at least for us, was that the vercel CLI had decided to change the project type that can be found in "Project Settings" in the GUI to "Next.js".  Changing it back to "Other" fixed the "error: not found: next" problem for us.

We believe that the vercel CLI decided to change this because one person in the team has been exploring ditching vercel in favour of moving our api routes in to next.js so we can rely on `next dev` directly.  This was obviously being done in a branch but was seemingly enough for vercel to decide to change the project type and break our mainline.  Another day lost for a team of three...

Good:
<img width="1043" alt="Screen Shot 2020-06-18 at 11 23 24 am" src="https://user-images.githubusercontent.com/814113/84967207-c514cd00-b156-11ea-98f6-4e52f413b8d2.png">

Bad:
<img width="1060" alt="Screen Shot 2020-06-18 at 11 23 37 am" src="https://user-images.githubusercontent.com/814113/84967226-d067f880-b156-11ea-8110-dcbf87c0e8f9.png">

**Reply by pranjaljately** [reply]
@dan-turner Great! Thanks, @dan-turner this worked for me as well.

**Comment by pranjaljately** [comment]
Although @dan-turner's method works, it feels more like a workaround than a fix.
